docs/source/howto/plugin_codes.rst

```diff
@@ -4,172 +4,188 @@
-The engine will store these files in a :py:class:`~aiida.orm.nodes.data.folder.FolderData` node that will be attached as an output node to the calculation with the label ``retrieved``.
+We want to pass our input file to the executable via standard input, and record standard output of the executable in the output file -- this is done using the ``stdin_name`` and ``stdout_name`` attributes of the |CodeInfo|.
+
+.. tip::
+
+     Many executables don't read from standard input but instead require the path to an input file to be passed via command line parameters (potentially including further configuration options).
+     In that case, use the |CodeInfo| ``cmdline_params`` attribute:
+
+     .. code-block:: python
+
+         codeinfo.cmdline_params = ['--input', self.inputs.input_filename]
+
+.. tip::
+
+    ``self.inputs.input_filename`` is just a shorthand for ``self.inputs.metadata['options']['input_filename']``.
+
+Finally, we pass the |CodeInfo| to a |CalcInfo| object (one calculation job can involve more than one executable, so ``codes_info`` is a list).
+We define the ``retrieve_list`` of filenames that the engine should retrieve from the directory where the job ran after it has finished.
+The engine will store these files in a |FolderData| node that will be attached as an output node to the calculation with the label ``retrieved``.
 There are :ref:`other file lists available<topics:calculations:usage:calcjobs:file_lists>` that allow you to easily customize how to move files to and from the remote working directory in order to prevent the creation of unnecessary copies.
 
-This was a minimal example of how to implement the |CalcJob| class to interface AiiDA with an external code.
-For more detailed information and advanced functionality on the |CalcJob| class, refer to the Topics section on :ref:`defining calculations <topics:calculations:usage>`.
+This was an example of how to implement the |CalcJob| class to interface AiiDA with an external code.
+For more details on the |CalcJob| class, refer to the Topics section on :ref:`defining calculations <topics:calculations:usage>`.
 
 .. _how-to:plugin-codes:parsing:
 
 Parsing the outputs
 ===================
 
-The parsing of the output files generated by a |CalcJob| is optional and can be used to store (part of) their information as AiiDA nodes, which makes the data queryable and therefore easier to access and analyze.
-To enable |CalcJob| output file parsing, one should subclass the |Parser| class and implement the :py:meth:`~aiida.parsers.parser.Parser.parse` method.
-The following is an example implementation, as shipped with ``aiida-core``, to parse the outputs of the :py:class:`~aiida.calculations.arithmetic.add.ArithmeticAddCalculation` discussed in the previous section:
+Parsing the output files produced by your code into AiiDA nodes is optional, but it can make your data queryable and therefore easier to access and analyze.
```

Comment:
```suggestion
Parsing the output files produced by a code into AiiDA nodes is optional, but it can make your data queryable and therefore easier to access and analyze.
```